grails-app/services/au/org/ala/ecodata/ProjectService.groovy

```diff
@@ -735,50 +746,48 @@ class ProjectService {
-                    try {
-                        sciStarterProjectUrl = "${grailsApplication.config.getProperty('scistarter.baseUrl')}${grailsApplication.config.getProperty('scistarter.projectUrl')}/${project.id}?key=${grailsApplication.config.getProperty('scistarter.apiKey')}"
-                        String text = webService.get(sciStarterProjectUrl, false);
-                        if (text instanceof String) {
-                            Map projectDetails = jsonSlurper.parseText(text)
-                            if (projectDetails.origin && projectDetails.origin == 'atlasoflivingaustralia') {
-                                // ignore projects SciStarter imported from Biocollect
-                                log.warn("Ignoring ${projectDetails.title} - ${projectDetails.id} - This is an ALA project.")
-                                ignoredProjects++
-                            } else {
-                                projectDetails << project
-                                // map properties from SciStarter to Biocollect
-                                transformedProject = SciStarterConverter.convert(projectDetails)
-                                if (!importedSciStarterProject) {
-                                    // create project & document & site & organisation
-                                    createSciStarterProject(transformedProject, projectDetails)
-                                    createdProjects++
-                                } else {
-                                    // update a project just in case something has changed.
-                                    updateSciStarterProject(transformedProject, importedSciStarterProject)
-                                    log.info("Updating ${importedSciStarterProject.name} ${importedSciStarterProject.projectId}.")
-                                    updatedProjects++
-                                }
-                            }
-                        }
-                    } catch (Exception e) {
-                        log.error("Error processing project - ${sciStarterProjectUrl}. Ignoring it. ${e.message}", e);
+
+            while(true) {
+                // list SciStarter projects for the current page
+                Map data = getSciStarterProjectsPage(slurper, page)
+                log.info("-- PAGE ${page}/${Math.round(data.total / 10)} SCISTARTER --")
+
+                // Break the loop if there are no more projects left to import
+                if (data.entities.size() == 0) break
+
+                data.entities.eachWithIndex { project, index ->
+                    Project existingProject = Project.findByExternalIdAndIsSciStarter(project.legacy_id.toString(), true)
+
+                    if (project.origin == 'atlasoflivingaustralia') {
+                        // ignore projects SciStarter imported from BioCollect
+                        log.warn("Ignoring ALA project ${project.name} - ${project.id}")
```

Comment:
Is this a warning or something we are expecting to happen every import?  I guess if we don't do it frequently it doesn't matter as it won't be filling up logs.

Reply:
This will happen every import - SciStarter contains projects scraped from BioCollect, so this check was added so that we're not re-importing projects that originated in BioCollect.